kuryr_kubernetes/controller/drivers/namespace_security_groups.py

```diff
@@ -109,6 +110,14 @@ def create_namespace_sg(self, namespace, project_id, crd_spec):
                     }
                 }).get('security_group')
             utils.tag_neutron_resources('security-groups', [sg['id']])
+            # NOTE(ltomasbo): Neutron populates every new SG with two rules
+            #                 allowing egress on IPv4 and IPv6. As namespace
+            #                 isolation does not handle egress, we remove them
+            #                 and leave egress for the default pods SG, or
+            #                 open to modifications per namespace.
+            for sgr in sg['security_group_rules']:
```

Comment:
If Kuryr dies in this moment the SG rules won't get deleted, but I guess in 3.11 we have more places when such issue can happen.